src/daft-logical-plan/src/optimization/rules/split_vllm.rs

```diff
@@ -0,0 +1,62 @@
+use common_error::DaftResult;
```

Comment:
This is fine for now, but I believe `granular_projections.rs` should handle a lot of the limitations automatically and it should be easy to hack in a translation to convert the split expression into a custom plan node. I did it for the URL operators in its PR.